### ISSUE TITLE
`experimentalExtensionlessBinWorkaround` option to resolve extensionless entrypoints to CJS in ESM loader

### ISSUE THREAD

**Issue by cspotcode** [TypeStrong/ts-node#1611]
This is labelled to indicate we will accept a pull request that implements this feature.

Intentionally marked experimental so that ts-node can remove or change it at any time without following semver.  Intentionally has verbose name to include the word "workaround."

When a user enables this and they use the ESM loader, it will resolve extensionless entrypoints to CJS format to workaround the errors caused by https://github.com/nodejs/node/issues/33226

This is an intentional divergence from node's behavior.  Users who enable this flag must understand that it may cause breakages in the future.  However, in the present, it will unblock those users, allowing them to combine ts-node/esm with third-party tools that have extensionless bin entrypoints.

Can be named either `experimentalExtensionlessBinWorkaround` or `experimentalExtensionlessEntrypointWorkaround`

See also: discussion starting here
https://github.com/TypeStrong/ts-node/issues/1007#issuecomment-1020033232

**Comment by cspotcode** [comment]
Closing in favor of #1649
